src/postgres/getClientConfig.ts

```diff
@@ -12,25 +12,33 @@ import { invalidCredentialsErrorType } from "./tree/PostgresDatabaseTreeItem";
 import { PostgresServerTreeItem } from "./tree/PostgresServerTreeItem";
 
 export async function getClientConfig(treeItem: PostgresServerTreeItem, databaseName: string): Promise<ClientConfig> {
-    let username: string | undefined = treeItem.connectionString.username;
-    let password: string | undefined = treeItem.connectionString.password;
+    let username: string = treeItem.connectionString.username;
+    let password: string = treeItem.connectionString.password;
 
     if (!(username && password)) {
         const credentials = await treeItem.getCredentials();
-        username = nonNullProp(credentials, 'username');
-        password = nonNullProp(credentials, 'password');
+        if (credentials.username && credentials.password) {
+            username = credentials.username;
+            password = credentials.password;
+        }
     }
 
-    const ssl: ConnectionOptions = {
+    const sslAzure: ConnectionOptions = {
         // Always provide the certificate since it is accepted even when SSL is disabled
         // Certificate source: https://aka.ms/AA7wnvl
         ca: BaltimoreCyberTrustRoot
     };
 
-    if (username && password) {
+    if ((username && password) || username === 'postgres') {
         const host = nonNullProp(treeItem.connectionString, 'hostName');
         const port: number = treeItem.connectionString.port ? parseInt(treeItem.connectionString.port) : postgresDefaultPort;
-        const clientConfig = { user: username, password: password, ssl, host, port, database: databaseName };
+        let ssl: boolean | ConnectionOptions | undefined;
+        if (treeItem.connectionString.hostName.endsWith(`.postgres.database.azure.com`)) {
```

Comment:
The problem is that for attaching azure accounts we need to use ssl = "BaltimoreCyber...". We can include this in the connection string but that would include a larger code change as we need to store the certificate and have to figure out an appropriate way to include that value in the connection string so that it would work with other postgres platforms. I would prefer that we handle it this way for this release and create an issue for this. I am guessing the the ending of the host name ".com" would differ for different countries. How about we check if the name contains the rest of the substring? 
What do you think?